test/short-option-combined-with-value.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+/* eslint max-len: 0 */
+
+const test = require('tape');
+const { parseArgs } = require('../index.js');
+
+test('when combine string short with plain text then parsed as value', (t) => {
+  const passedArgs = ['-aHELLO'];
+  const passedOptions = { alpha: { short: 'a', type: 'string' } };
+  const expected = { flags: { alpha: true }, values: { alpha: 'HELLO' }, positionals: [] };
+
+  const result = parseArgs({ args: passedArgs, options: passedOptions });
+
+  t.deepEqual(result, expected);
+  t.end();
+});
+
+test('when combine low-config string short with plain text then parsed as value', (t) => {
+  const passedArgs = ['-aHELLO'];
+  const passedOptions = { a: { type: 'string' } };
+  const expected = { flags: { a: true }, values: { a: 'HELLO' }, positionals: [] };
+
+  const result = parseArgs({ args: passedArgs, options: passedOptions });
+
+  t.deepEqual(result, expected);
+  t.end();
+});
+
+test('when combine string short with value like short option then parsed as value', (t) => {
+  const passedArgs = ['-a-b'];
```

Comment:
These are from the hard-core Open Group guidelines, which does not even mention long options. But I find useful to understand the origins and underpinnings of many command-line parsing details.

https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html#tag_12_01

(Emphasis on last sentence added by me.)

> 12.1 Utility Argument Syntax
>...
>2. Option-arguments are shown separated from their options by <blank> characters, except when the option-argument is enclosed in the '[' and ']' notation to indicate that it is optional. This reflects the situation in which an optional option-argument (if present) is included within the same argument string as the option; for a mandatory option-argument, it is the next argument. The [Utility Syntax Guidelines](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html#tag_12_02) in Utility Syntax Guidelines require that the option be a separate argument from its option-argument and that option-arguments not be optional, but there are some exceptions in POSIX.1-2017 to ensure continued operation of historical applications:
>
>a. If the SYNOPSIS of a standard utility shows an option with a mandatory option-argument (as with [ -c option_argument] in the example), a conforming application shall use separate arguments for that option and its option-argument. *However, a conforming implementation shall also permit applications to specify the option and option-argument in the same argument string without intervening <blank> characters.*

-----

https://pubs.opengroup.org/onlinepubs/9699919799/functions/getopt.html

> The description has been written to make it clear that getopt(), like the getopts utility, deals with option-arguments whether separated from the option by <blank> characters or not. Note that the requirements on getopt() and getopts are more stringent than the Utility Syntax Guidelines.

-----

https://pubs.opengroup.org/onlinepubs/9699919799/xrat/V4_xbd_chap12.html

SYNOPSIS Shows: `-a arg`
Conforming application uses: `-a arg`
System supports: `-a arg` and `-aarg`
Non-conforming applications may use: `-aarg`

Reply:
Very helpful reference! And reading further answered another question I had regarding what characters are allowed in option arguments:
> Guideline 7 allows any string to be an option-argument; an option-argument can begin with any character, can be `-` or `--`, and can be an empty string. For example, the commands `pr -h -`, `pr -h --`, `pr -h -d`, `pr -h +2`, and `pr -h "` contain the option-arguments `-`, `--`, `-d`, `+2`, and an empty string, respectively. Conversely, the command `pr -h -- -d` treats `-d` as an option, not as an argument, because the `--` is an option-argument here, not a delimiter.

Almost makes me wonder if we should always capture the `nextArg` if `options[longOption].type === 'string'`. Perhaps we shouldn't open that can of worms 😅 I'm quite satisfied with the current behavior.

Reply:
Yes indeed, glad you mentioned it! I have actually been thinking about this and doing some research this week. I opened #76 partly to provide a context for revisiting whether options are "greedy", and was leaning towards opening a new issue about that. Now you have raised it, I definitely will!

Before I was clear on the implications I had opened #25 which covers this ground too, but much of the discussion there is about what to store in `strict:false` mode if the value is missing, and not deep coverage of greedy vs non-greedy.

I'm ok with shipping either way, but would like people to be clearer on where the behaviour stands in comparison with other implementations and why.

Reply:
> would like people to be clearer on where the behavior stands in comparison with other implementations and why.

Yes! I think it would be super helpful to have a clear understanding/consensus on what standards and implementations we follow and/or influence the `parseArgs` API.